app/settings/developer-apps/edit/route.ts

```diff
@@ -15,6 +15,7 @@ export default class SettingsDeveloperAppsEditRoute extends Route.extend({
 }) {
     @service router!: RouterService;
 
+    // eslint-disable-next-line camelcase
     model(params: { developer_app_id: string }) {
```

Comment:
shouldn't this be camel case though?

Reply:
It really should IMHO. Not sure why we went with snake case for this. It comes from here:
https://github.com/CenterForOpenScience/ember-osf-web/blob/d52674e1d6a4cacd64dbd4f2cd7be12e094f24c3/app/router.ts#L110
@aaxelb, since you added this route, thoughts?

Reply:
I'm guessing I was following the [convention in the ember guides](https://guides.emberjs.com/v3.7.0/routing/defining-your-routes/#toc_dynamic-segments) without thinking about it. We're not using the automatic model hooks (and if we were it should be `:developer-app_id`), so there's no reason not to switch to camel case.

Reply:
So I tried changing the dynamic segments to be camelCase, but this violates:
[ember/routes-segments-snake-case](https://github.com/ember-cli/eslint-plugin-ember/blob/master/docs/rules/routes-segments-snake-case.md)

But actually, it turns out that we [should be disabling the `camelcase` rule in favor of `@typescript-eslint/camelcase`](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/camelcase.md#options). This does indeed make this line acceptable (because it's defining a type, not an object as the `camelcase` rule thinks), but it blows up **lots** of other things:

`✖ 653 problems (653 errors, 0 warnings)`

Most of that is:

```
$ npx eslint app/locales/en/translations.ts

✖ 515 problems (515 errors, 0 warnings)
```
But that still leaves 138 other issues that are sprinkled throughout the codebase.

There seems to be inconsistencies between `camelcase` and  `@typescript-eslint/camelcase` WRT property and class field naming. We should investigate this, but, IMHO, that's beyond the scope of this PR (and I don't want to blow it up any more). I think we should just leave it disabled here for now and then look at `@typescript-eslint/camelcase` in another PR.

Reply:
Can you open up that PR now and add those changes for camel case, otherwise we might forget or not review it. And is translations complaining because we don't use camel case for a lot of that file? TBH that makes a lot of sense - that file needed to be refactored since it mixes case inside of it anyways.